storage/range_command.go

```diff
@@ -337,21 +340,26 @@ func (r *Range) InternalRangeLookup(batch engine.Engine, args *proto.InternalRan
 			"Range lookup specified invalid maximum range count %d: must be > 0", rangeCount))
 		return nil
 	}
-	if args.IgnoreIntents {
-		rangeCount = 1 // simplify lookup because we may have to retry to read new
+	consistent := args.ReadConsistency != proto.INCONSISTENT
+	if consistent && args.IgnoreIntents {
+		reply.SetGoError(util.Errorf("can not read consistently and skip intents"))
+		return nil
+	}
+	if args.IgnoreIntents || consistent {
```

Comment:
There's no point in prefetching here except for running into more intents and failing more often. I agree that it's orthogonal but the optimization really only makes sense if there's someone who caches, which isn't the case here. So by default, you get your descriptor, and everything else is bells and whistles we can add in later should we need them.